x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/ttest/TTestAggregatorFactory.java

```diff
@@ -0,0 +1,81 @@
+                           AggregatorFactories.Builder subFactoriesBuilder,
+                           Map<String, Object> metadata) throws IOException {
+        super(name, configs, format, queryShardContext, parent, subFactoriesBuilder, metadata);
+        this.testType = testType;
+        this.tails = tails;
+    }
+
+    @Override
+    protected Aggregator createUnmapped(SearchContext searchContext,
+                                        Aggregator parent,
+                                        List<PipelineAggregator> pipelineAggregators,
+                                        Map<String, Object> metadata) throws IOException {
+        switch (testType) {
+            case PAIRED:
+                return new PairedTTestAggregator(name, null, tails, format, searchContext, parent, pipelineAggregators, metadata);
+            case HOMOSCEDASTIC:
+                return new UnpairedTTestAggregator(name, null, tails, true, format, searchContext, parent, pipelineAggregators, metadata);
+            case HETEROSCEDASTIC:
+                return new UnpairedTTestAggregator(name, null, tails, false, format, searchContext, parent, pipelineAggregators, metadata);
+            default:
+                throw new UnsupportedOperationException("Unsupported t-test type " + testType);
+        }
+    }
+
+    @Override
+    protected Aggregator doCreateInternal(SearchContext searchContext,
+                                          Map<String, ValuesSourceConfig> configs,
+                                          DocValueFormat format,
+                                          Aggregator parent,
+                                          boolean collectsFromSingleBucket,
+                                          List<PipelineAggregator> pipelineAggregators,
+                                          Map<String, Object> metadata) throws IOException {
+        MultiValuesSource.NumericMultiValuesSource numericMultiVS
+            = new MultiValuesSource.NumericMultiValuesSource(configs, queryShardContext);
+        if (numericMultiVS.areValuesSourcesEmpty()) {
+            return createUnmapped(searchContext, parent, pipelineAggregators, metadata);
+        }
+        switch (testType) {
+            case PAIRED:
+                return new PairedTTestAggregator(name, numericMultiVS, tails, format, searchContext, parent, pipelineAggregators, metadata);
+            case HOMOSCEDASTIC:
+                return new UnpairedTTestAggregator(name, numericMultiVS, tails, true, format, searchContext, parent, pipelineAggregators,
+                    metadata);
+            case HETEROSCEDASTIC:
+                return new UnpairedTTestAggregator(name, numericMultiVS, tails, false, format, searchContext, parent, pipelineAggregators,
+                    metadata);
+            default:
+                throw new UnsupportedOperationException("Unsupported t-test type " + testType);
```

Comment:
Hmm, do we know if UnsupportedOpException is a 4xx or 5xx? If it's a 5xx, perhaps we should change this to an IllegalArgumentException?